Parser/action_helpers.c

```diff
@@ -1635,14 +1634,17 @@ _PyPegen_concatenate_strings(Parser *p, asdl_expr_seq *strings,
                    "abc" u"abc" ->  "abcabc" */
                 PyObject *kind = elem->v.Constant.kind;
 
-                _PyUnicodeWriter_Init(&writer);
+                PyUnicodeWriter *writer = PyUnicodeWriter_Create(0);
```

Comment:
What's the 0 here?

Reply:
IIRC that's the number of pre-allocated characters. If you know that you will need N characters, I think it's better to put N in advance so that you don't have to resize the buffer. Otherwise, 0 is a good choice if you don't know in advance how many characters you'll eventually need (at the cost of resizes).